### ISSUE TITLE
session 3. neural network

### ISSUE THREAD

**Issue by Odin-son** [Odin-son/felis-catus#1]

**Comment by Odin-son** [comment]
2020.11.02

**Comment by Odin-son** [comment]
퍼셉트론에서 신경망으로,
계단함수에서 시그모이드함수로.

- 신경망에서 활성화함수로 계단함수(e.g. y=cx)를 사용한다면, 은닉층을 거친 결과가 결국은 선형함수(e.g. y=(c^3)x)가 되기 때문에 의미가 없다.

- 활성화 함수
  - Sigmoid func. 1 / (1+np.exp(-x))
  - ReLU func. np.maximum(0, x)

- 출력층의 활성화 함수 (notation sigma)
   - Identity func.
   - softmax func. np.exp(a) / np.sum(np.exp(a)) => overflow를 막기 위해, c=np.max(a), np.exp(a-c) / np.sum(np.exp(a-c)) 

- 정규화

**Comment by Odin-son** [comment]
회귀 문제에는 항등함수를 쓰며, 분류 문제에는 소프트맥스 함수를 씀
